### PR TITLE
More test for sortby

### PR DESCRIPTION
**Pull request by milg15**
Tests for either key, when the values are the same or there is an empty space. Is there another case?

(Solves #31)

### CODE REVIEW

File: packages/noya-utils/src/__tests__/sortBy.test.ts [hunk 1]
```diff
@@ -3,3 +3,83 @@ import { sortBy } from '../index';
 test('empty array', () => {
   expect(sortBy([], '')).toEqual([]);
 });
+
+test('first key', () => {
+  expect(
+    sortBy(
+      [
```
Comment (dabbott) [inline packages/noya-utils/src/__tests__/sortBy.test.ts:10]: If this is an array we use multiple times, we can move it outside of the `test` function to reuse it.

File: packages/noya-utils/src/__tests__/sortBy.test.ts [hunk 2]
```diff
@@ -3,3 +3,83 @@ import { sortBy } from '../index';
...
+    { name: 'Emma', last: 'Jones' },
+    { name: 'Liam', last: 'Smith' },
+  ]);
+});
+
+test('same value', () => {
+  expect(
+    sortBy(
+      [
+        { name: 'Liam', last: 'Smith' },
```
Comment (dabbott) [inline packages/noya-utils/src/__tests__/sortBy.test.ts:53]: At this point we've already tested different keys, so we can probably remove `last` to keep it simpler

File: packages/noya-utils/src/__tests__/sortBy.test.ts [hunk 3]
```diff
@@ -3,3 +3,83 @@ import { sortBy } from '../index';
...
+    { name: 'Liam', last: 'Smith' },
+    { name: 'Liam', last: 'Brown' },
+  ]);
+});
+
+test('empty space', () => {
+  expect(
+    sortBy(
+      [
+        { group: 'cloud', ammount: 9 },
```
Comment (dabbott) [inline packages/noya-utils/src/__tests__/sortBy.test.ts:70]: I think we can remove `ammount` here. And shorten the array to just 2 items, one with an empty string and one without

**Review by dabbott — CHANGES_REQUESTED** [review]
Let's add

- A test that clarifies how we handle uppercase and lowercase comparison, e.g. ["A", "a"]
- A test that contains some "/" since we compare paths with this, e.g. ["a", "aa/b", "a/b/c", "a/b"]

For both of these we probably only need a single a key in the objects we compare, to keep it short

File: packages/noya-utils/src/__tests__/sortBy.test.ts (hunk 2 above)
Comment (dabbott) [reply packages/noya-utils/src/__tests__/sortBy.test.ts:53]: Actually I guess we wouldn't be able to tell them apart without `last`...

Comment (milg15) [reply packages/noya-utils/src/__tests__/sortBy.test.ts:53]: so I keep them both? or only the first one?

Comment (dabbott) [reply packages/noya-utils/src/__tests__/sortBy.test.ts:53]: Let's keep it the way it is now

File: packages/noya-utils/src/__tests__/sortBy.test.ts (hunk 1 above)
Comment (milg15) [reply packages/noya-utils/src/__tests__/sortBy.test.ts:10]: where would we place this array?

Comment (dabbott) [reply packages/noya-utils/src/__tests__/sortBy.test.ts:10]: Outside of every function, so e.g.

```ts
const names = [{ name: 'Liam' }, ...]

test('first key', () => {
  expect(sortBy(names, 'name'))...
})

test('second key', () => {
  expect(sortBy(names, 'last'))...
})
```

File: packages/noya-utils/src/__tests__/sortBy.test.ts [hunk 4]
```diff
@@ -3,3 +3,89 @@ import { sortBy } from '../index';
...
+    { letter: 'A' },
+    { letter: 'B' },
+    { letter: 'b' },
+  ]);
+});
+
+test('"/" usage', () => {
+  expect(
+    sortBy(
+      [{ group: 'B/a' }, { group: 'A/d' }, { group: 'B/c' }, { group: 'A/n' }],
```
Comment (dabbott) [inline packages/noya-utils/src/__tests__/sortBy.test.ts:82]: In this case I want to test comparison between letters, like "B", and "/". Right now the "/" is always in the same position, so not testing that. E.g. we should have "A/BB" and "AA/B" or something.

(Thinking about this some more, we probably can't sort lists of paths using this method way we do in one of the grid components, but we can come back to that another time)

**Review by dabbott — CHANGES_REQUESTED** [review]

File: packages/noya-utils/src/__tests__/sortBy.test.ts (hunk 1 above)
Comment (milg15) [reply packages/noya-utils/src/__tests__/sortBy.test.ts:10]: Got it!

**Review by dabbott — APPROVED** [review]

**Pull request merged**